source/extensions/filters/network/kafka/mesh/abstract_command.h

```diff
@@ -0,0 +1,97 @@
+#pragma once
+
+#include "source/common/common/logger.h"
+#include "source/extensions/filters/network/kafka/kafka_response.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace NetworkFilters {
+namespace Kafka {
+namespace Mesh {
+
+/**
+ * Represents single downstream client request.
+ * Responsible for performing the work on multiple upstream clusters and aggregating the results.
+ */
+class InFlightRequest {
```

Comment:
implementations of this interface will provide the proper handling to provide kafka-mesh features
to have basic compliance with protocol we will require to handle the following types
* api versions - [to tell client we understand some requests](https://github.com/envoyproxy/envoy/pull/11936/files#diff-1c8b61805997d18486e049aa302713356f53a6fc53eb6ecd6ba29fdbf41aca1bR29) 
* metadata - [to tell client to send produce requests to our envoy instance](https://github.com/envoyproxy/envoy/pull/11936/files#diff-6f031aa07a0ebd92856b2d409255e4a043842edb60a28cb5d8f916c4e78e110aR27)
* produce - [to actually process the data and send them upstream to kafka](https://github.com/envoyproxy/envoy/pull/11936/files#diff-a5b85c101e48a968f816536f17e9e7ab84db7abf19e468826f30085260ad2421R23)